---
title: Docs
template: blank
---

[Source](http://atlas.gs.washington.edu/fly-atac/ 'Permalink to A Chromatin Cell Atlas of the Developing Fly Embryo')

# A Chromatin Cell Atlas of the Developing Fly Embryo

In [Cusanovich, Reddington, Garfield _et al._ (_Nature_, 2018)][1], we reported single-cell ATAC-seq of _Drosophila melanogaster_ embryos at 3 different stages of development (2-4, 6-8, and 10-12 hours after egg laying) using a combinatorial indexing approach (sci-ATAC-seq).

Here, you can download some of the data for yourself and explore how cells clearly cluster into the germ layers by 6-8 hours after egg laying.

[Get started Â»][2]

We have built this site as a companion to the resources that can be found on the [Furlong lab's website][3]. On that site, the Furlong group has produced a genome browser and several other tools so that you can interact with the data visually. We encourage you to have a look to get a sense of the kind of data available! On this site, we hope to give you a sense of the underlying data structures and how you can analyze it yourself. The [tutorial][2] section describes how to download some of the data and install required software. In addition, we provide some examples of the types of analyses that were presented in the paper. If you want to dig even further into the data, you can visit our [GitHub repository][4], where we provide instructions for processing the raw sequencing data.

[1]: https://www.nature.com/articles/nature25981
[2]: http://atlas.gs.washington.edu/fly-atac/docs/
[3]: http://shiny.furlonglab.embl.de/scATACseqBrowser/
[4]: https://github.com/shendurelab/fly-atac/
